shards/core/runtime.hpp

```diff
@@ -246,6 +246,222 @@ struct DefaultHelpText {
   static inline const SHOptionalString OutputHelpPass = SHCCSTR("Outputs the input value, passed through unchanged.");
 };
 
+// LayeredMap Class Definition
+template <typename K, typename V> class LayeredMap {
+  std::vector<std::unordered_map<K, V>> layers;
+  std::vector<bool> blocker_tags;
+  size_t first_blocker_index = std::numeric_limits<size_t>::max(); // Index of the first blocker
```

Comment:
What does this do? shouldn't searching from back to front and stop on blocker tag work for all cases?

Reply:
Ah eh, didn't check, blame AI 🤖🤣